### PR TITLE
More Jira -> Issues doc updates

### PR DESCRIPTION
**Pull request by damccorm**
This is a follow up to #21749 - it fixes a couple inconsistencies/mistakes introduced there and also makes a few additional updates.

Part of #21686

-----------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Add a link to the appropriate issue in your description, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:
 
  - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
- - [ ] Add a link to the appropriate issue in your description, if applicable. This will automatically link the pull request to the issue.
+ - [ ] Mention the appropriate issue in your description (for example: "addresses #123), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment "fixes #<ISSUE NUMBER>" instead.
```
Comment (Abacn) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: "automatically close ... comment "fixes #<ISSUE NUMBER>" That's nice. Is this already implemented?

Comment (damccorm) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: Yeah, that's a GitHub native feature

**Comment by damccorm** [comment]
R: @aijamalnk

**Comment by aijamalnk** [comment]
Looks great, thank you!

**Pull request merged**